### ISSUE TITLE
ERROR: cannot use ML routines

### ISSUE THREAD

**Issue by wangleidnadna** [freesurfer/freesurfer#1274]
Hello FreeSurfer Developers,

Here are my problems
-------- freesurfer-macOS-darwin_arm64-8.0.0-beta-20241103-b8aacdc --------
Setting up environment for FreeSurfer/FS-FAST (and FSL)
FREESURFER_HOME   /Applications/freesurfer/8.0.0-beta
FSFAST_HOME       /Applications/freesurfer/8.0.0-beta/fsfast
FSF_OUTPUT_FORMAT nii.gz
SUBJECTS_DIR      /Applications/freesurfer/8.0.0-beta/subjects
MNI_DIR           /Applications/freesurfer/8.0.0-beta/mni
FSL_DIR           /Users/wanglei/fsl
wanglei@wangleideMacBook-Pro T1Img % pwd
/Users/wanglei/Downloads/DemoData/DemoRaw/DemoData/T1Img
wanglei@wangleideMacBook-Pro T1Img % recon-all -s subj_01 -i subj_01_t1.nii -all
Adding options -fix-ento-wm -transfer-base-bfs -fix-vsinus -fix-mca-dura -fix-ga -fix-acj -synthstrip -synthseg -synthmorph 
ERROR: cannot use ML routines

**Comment by ElChiriquiski** [comment]
I have the same issue with V8 beta:

recon-all -s 3102 -i *.nii.gz -all -no-isrunning
Adding options -fix-ento-wm -transfer-base-bfs -fix-vsinus -fix-mca-dura -fix-ga -fix-acj -synthstrip -synthseg -synthmorph 
ERROR: cannot use ML routines

Could it be because I have it built within a docker container?

**Comment by dngreve** [comment]
You must set environment variable FS_ALLOW_DEEP to 1
See https://surfer.nmr.mgh.harvard.edu/fswiki/ReleaseNotes


On 1/16/2025 4:03 PM, ElChiriquiski wrote:
>
>         External Email - Use Caution
>
> I have the same issue with V8 beta:
>
> recon-all -s 3102 -i *.nii.gz -all -no-isrunning
> Adding options -fix-ento-wm -transfer-base-bfs -fix-vsinus 
> -fix-mca-dura -fix-ga -fix-acj -synthstrip -synthseg -synthmorph
> ERROR: cannot use ML routines
>
> —
> Reply to this email directly, view it on GitHub 
> <https://secure-web.cisco.com/17XHbLcJZ9QgIsy5dSV8sjgXNWgPBMS-ShRdxZlCo0KIC_TR48cvmdXAwztyXcBE110XWyllkw96wuZJNCWsUkY35pPesMna3cyP96Q4t7032QfCtXeQ3Iic9EWUXKwZO9ayJsot2M78Y54l1KbB35JrQMaetmVHzjbgDMRkfCGPEY6LqTCTRqQqY55DBuTey8BfINpvqgSSMtYwf7TjzCcvo5NZWRewxGYcJvv2v4fuR6sue3EI_25T1sXItJnrEmyuJ2TYcDTn6yACX6oyCHlcqe3tJqBcbpvaT0zlatBcNE__CE5KeFykt5w9JsiRs/https%3A%2F%2Fgithub.com%2Ffreesurfer%2Ffreesurfer%2Fissues%2F1274%23issuecomment-2596883270>, 
> or unsubscribe 
> <https://secure-web.cisco.com/1LHBde3XMcFhVFEoJ8VK0tci5N05UI95cvLvuYMXqy8t1hpC1x_O0-6ahMpl3EXw5s7LCJUNBtHTa5mtGLLwn3CHS8QyYs_XR2llxSdjVkfsQtOIAlTCrB7-Nh0PFXJ39o8vTPkyYPIhPp6TEew1cQzrlllAm2JoIsKWVcPNp-rMKLQ-jFaNISLHNXBgkP4BdrEMMfTulJQbvc_4nz2rVA72DPHT4C67dBgFmy3r-H9eXZLgCCpHu1YIoYaoIWx9XDrMEjguTRISi9vhR4VTJEyujyYXOgVOUnP_9kaT6or7o76Y-6dsY7jU3WK6738ST/https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FAGIX2VUJW7233QG73OXOX332LANCNAVCNFSM6AAAAABVCYKRPOVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDKOJWHA4DGMRXGA>.
> You are receiving this because you are subscribed to this 
> thread.Message ID: 
> ***@***.***>
>
The information in this e-mail is intended only for the person to whom it is addressed.  If you believe this e-mail was sent to you in error and the e-mail contains patient information, please contact the Mass General Brigham Compliance HelpLine at https://www.massgeneralbrigham.org/complianceline <https://www.massgeneralbrigham.org/complianceline> .
Please note that this e-mail is not secure (encrypted).  If you do not wish to continue communication over unencrypted e-mail, please notify the sender of this message immediately.  Continuing to send or respond to e-mail after receiving this message means you understand and accept this risk and wish to continue to communicate over unencrypted e-mail.

**Comment by wangleidnadna** [comment]
Thank you

**Comment by wangleidnadna** [comment]
[recon-all.log](https://github.com/user-attachments/files/18471705/recon-all.log)

Thanks. Here is another problem


 File "/Applications/freesurfer/8.0.0-beta/python/miniconda3_apple_dist/py3_8_13/lib/python3.8/site-packages/keras/src/utils/control_flow_util.py", line 108, in smart_cond
      return tf.__internal__.smart_cond.smart_cond(
    File "/Applications/freesurfer/8.0.0-beta/python/miniconda3_apple_dist/py3_8_13/lib/python3.8/site-packages/keras/src/layers/normalization/batch_normalization.py", line 979, in _fused_batch_norm_inference
      return tf.compat.v1.nn.fused_batch_norm(
Node: 'unet/unet_bn_down_0/FusedBatchNormV3'
2 root error(s) found.
  (0) INVALID_ARGUMENT:  input must be 4-dimensional[1,72,72,72,24]
	 [[{{node unet/unet_bn_down_0/FusedBatchNormV3}}]]
	 [[unet/unet_merge_3/concat/_48]]
  (1) INVALID_ARGUMENT:  input must be 4-dimensional[1,72,72,72,24]
	 [[{{node unet/unet_bn_down_0/FusedBatchNormV3}}]]
0 successful operations.
0 derived errors ignored. [Op:__inference_predict_function_1146]
ERROR:
Darwin wangleideMacBook-Pro.local 23.4.0 Darwin Kernel Version 23.4.0: Wed Feb 21 21:44:54 PST 2024; root:xnu-10063.101.15~2/RELEASE_ARM64_T6031 arm64

recon-all -s subj_01 exited with ERRORS at Mon Jan 20 10:10:07 CST 2025

For more details, see the log file /Users/wanglei/Downloads/DemoData/DemoRaw/DemoData/subj_01/subj_01/scripts/recon-all.log
To report a problem, see http://surfer.nmr.mgh.harvard.edu/fswiki/BugReporting

**Comment by ElChiriquiski** [comment]
Thank you @dngreve this worked!

**Comment by paulweger** [comment]
@wangleidnadna I got the same error with the Invalid argument / shape mismatch. Did you find a solution yet?

**Comment by yhuang43** [comment]
This is fixed in the coming new 8.0.0 release (commit https://github.com/freesurfer/freesurfer/commit/8092c35311ed2430a975d9c0336a4146158e187e).

**Comment by yhuang43** [comment]
The recent FS 8.0.0 release fixed the bug - https://surfer.nmr.mgh.harvard.edu/fswiki/rel7downloads
Let us know if you are still experiencing the issue with the new release.
